enhancements/api-review/add-ImagePolicy-to-config.openshift.io-v1alpha1.md

```diff
@@ -0,0 +1,294 @@
+---
+title: add-ImagePolicy-to-config.openshift.io-v1alpha1
+authors:
+  - "@QiWang19"
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - "@saschagrunert, for API design, implementation details and graduation criteria"
+  - "@mrunalp"
+  - "@yuqi-zhang"
+  - "@mtrmac"
+approvers: # A single approver is preferred, the role of the approver is to raise important questions, help ensure the enhancement receives reviews from all applicable areas/SMEs, and determine when consensus is achieved such that the EP can move forward to implementation.  Having multiple approvers makes it difficult to determine who is responsible for the actual approval.
+  - "@mrunalp"
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - "@mrunalp"
+creation-date: 2023-05-17
+last-updated: yyyy-mm-dd
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/OCPNODE-1628
+---
+
+# Add CRD ImagePolicy to config.openshift.io/v1alpha1
+
+## Summary
+
+This enhancement propose an ImagePolicy CRD to provide a way to set sigstore image signature verification configuration on Openshift.
+
+## Motivation
+
+### User Stories
+
+As an OpenShift user, I want to verify the container images signed using sigstore
+tools, so that I can utilize the increased security of my software supply chain.
+
+### Goals
+
+- The user can create an ImagePolicy instance specifying the images/repositories to be verified and their policy. The MCO will write the configuration for signature verification. Once this is done, CRI-O can verify the images/repositories.
+
+- MCO container runtime config controller watches ImagePolicy instance in different namespaces and merges the instances for each namespace into a single [containers-policy.json](https://github.com/containers/image/blob/main/docs/containers-policy.json.5.md) in a predefined `/path/to/policies/\<NAMESPACE\>.json`.
+
+- CRI-O can verify Cosign signature signed images using configuration from ImagePolicy by using the namespace from the sandbox config on the `PullImage` RPC.
+
+### Non-Goals
```

Comment:
Some of the entries from *Workflow Description* can be bubbled up here as explicit non-goals.  For example:

* Do not grant the application administrator the ability to weaken cluster-scoped policies, to avoid expanding the set of administrators capable of increasing cluster exposure to vulnerable images.
* Do not grant the application administrator  the ability to tighten cluster-scoped policies.  This could be useful in the future, but we are deferring it to limit the amount of work needed for an initial implementation.
* Etc., etc.